internal/handlers/common/update.go

```diff
@@ -328,6 +334,45 @@ func processIncFieldExpression(doc *types.Document, updateV any) (bool, error) {
 	return changed, nil
 }
 
+// processMaxFieldExpression changes document according to $max operator.
+// If the document was changed it returns true.
+func processMaxFieldExpression(doc *types.Document, updateV any) (bool, error) {
+	maxExpression := updateV.(*types.Document)
+
+	var changed bool
+
+	for _, field := range maxExpression.Keys() {
+		val, _ := doc.Get(field)
+
+		maxVal, err := maxExpression.Get(field)
+		if err != nil {
+			// if max field does not exist, don't change anything
+			continue
+		}
+
+		// if the key was not found, set it
```

Comment:
The comment doesn’t describe what’s happening in that if statement. Inside that if we are calling Compare and analyse compare results.

Reply:
But it describes what's happening when condition is not met and the whole motivation behind this if statement. But I can rephrase it to "if the document value was found, compare it with max value"